source/types/numeric.go

```diff
@@ -0,0 +1,58 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"github.com/jackc/pgx/v5/pgtype"
+)
+
+type NumericFormatter struct {
+	Value pgtype.Numeric
+}
+
+// Format coerces `pgtype.Numeric` to int or double depending on the exponent.
+// Returns error when value is invalid.
+func (n NumericFormatter) Format() (any, error) {
```

Comment:
Nitpick: if the function should always return an int or a float, maybe we can use generics to indicate that. Also, if time allows, would be good to add some unit tests for this.

Reply:
Fixed in last refactor.